pallets/pips/src/lib.rs

```diff
@@ -1025,7 +1025,7 @@ decl_module! {
             <SnapshotQueue<T>>::set(<LiveQueue<T>>::get());
 
             // Emit event.
-            Self::deposit_event(RawEvent::SnapshotTaken(did, id));
+            Self::deposit_event(RawEvent::SnapshotTaken(did, id, <SnapshotQueue<T>>::get()));
```

Comment:
More efficient:
```rust
             let queue = <LiveQueue<T>>::get();
             <SnapshotQueue<T>>::set(queue.clone());

             // Emit event.
             Self::deposit_event(RawEvent::SnapshotTaken(did, id, queue));
```

Reply:
If we don't need the weights we could also do the following, and get better storage efficiency:
```rust
             Self::deposit_event(RawEvent::SnapshotTaken(did, id, queue.into_iter().map(|x| x.id).collect());
```
(emit the event first to avoid the `queue.clone()`)

Reply:
I'd rather provide the weights in the event in case they are useful to the UI. I've made the first fix.